src/Components/Endpoints/src/Discovery/AssemblyComponentLibraryDescriptor.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Components.Discovery;
+
+/// <summary>
+/// Represents an assembly along with the components and pages included in it.
+/// </summary>
+/// <remarks>
+/// This API is meant to be consumed in a source generation context.
+/// </remarks>
+[DebuggerDisplay($"{{{nameof(GetDebuggerDisplay)}(),nq}}")]
+public class AssemblyComponentLibraryDescriptor
+{
+    /// <summary>
+    /// Initializes a new instance of <see cref="AssemblyComponentLibraryDescriptor"/>.</summary>
+    /// <param name="name">The assembly name.</param>
+    /// <param name="pages">The list of pages in the assembly.</param>
+    /// <param name="components">The list of components in the assembly.</param>
+    public AssemblyComponentLibraryDescriptor(string name, IReadOnlyList<PageComponentBuilder> pages, IReadOnlyList<ComponentBuilder> components)
```

Comment:
```suggestion
    public AssemblyComponentLibraryDescriptor(string assemblyName, IReadOnlyList<PageComponentBuilder> pages, IReadOnlyList<ComponentBuilder> components)
```

Feel free to ignore if you're not changing anything else - just a minor nit, not important.

Reply:
Actually never mind. I'm sure we'll do API review later.

Reply:
I'll change this when I update the source generator API